src/core/importType.js

```diff
@@ -26,11 +25,17 @@ export function isBuiltIn(name, settings, path) {
 
 function isExternalPath(path, name, settings) {
   const folders = (settings && settings['import/external-module-folders']) || ['node_modules']
+  return !path || folders.some(folder => isSubpath(folder, path))
+}
 
-  // extract the part before the first / (redux-saga/effects => redux-saga)
-  const packageName = name.match(/([^/]+)/)[0]
+function isSubpath(subpath, path) {
+  const subpathRegExpSrc = escapeRegExp(subpath.replace(/[/]$/, ''))
```

Comment:
i'm pretty concerned here about trying to manually regex escape and make a dynamic regex. is there a more reliable means of doing this test?

Reply:
There definitely is — I can do manual parsing and checking in a loop. I still don't think that the current way is unreliable, but I see how the more imperative solution might get better readability, so I'll go ahead with it.

Just in case, how the current way works:

1. Remove the trailing slash in case it's presented in an item.
2. In the case the item contains special regex symbols, [escape them](https://stackoverflow.com/a/3561711/804678).
3. Make a regex that checks if a path contains the item as a complete subpath:
3.a. If the item starts with a slash, make a regex that checks if a path contains the item between the start of the string and a path segment boundary (either end of the string or a slash).
3.b. Otherwise, make a regex that checks if a path contains the item between path segment boundaries (either start/end of the string or a slash).

But I see how this can be pretty unreadable from the code, so I'll re-write this using a loop instead.